### DISCUSSION TITLE
Can I prevent CSS from loading in the content script?

### DISCUSSION THREAD

**Discussion by wallw-dev** [aklinker1/vite-plugin-web-extension#122] [Q&A]
I have a content script which needs to just load its JS, and then a portion of it will add an iframe to the page and subsequently load the styles only inside that iframe (which are imported into app and sub-component definitions directly). Currently, the build correctly spits out `dist/style.css` and I have it entered in `web_accessible_resources`, but the resulting `dist/manifest.json` gets the following:

```json
{
  "content_scripts": [
    {
      "js": [
        "src/content.js"
      ],
      "css": [
        "style.css"
      ]
    }
  ]
}
```
which leaks styles expected to be contained by the iframe onto the whole page. Is there a way to avoid including the `"css"` entry there?

**Comment by aklinker1** [comment]
It should only include a CSS file in your manifest's content script if you import the CSS file into the content script at some point.

Are you including an HTML file in the plugin's `additionalInputs` option to use as the iframe's src, or are you creating it via a string? If you're using an HTML file, you should import the CSS file there instead of your content script, and everything will work.

If you could share a small reproduction, I could give you better advice. But yeah, it seems like you're importing the CSS when you shouldn't be.

**Reply by wallw-dev** [reply]
Technically it is imported by the content script, and then the script does something like this:

```typescript
// ...
import Frame from 'react-iframe-component'

// ...
return <Frame
  head={[
    <link key="1" rel="stylesheet" href={chrome.runtime.getURL('style.css')}/>
  ]}
>
  <App/> {/* this component and some of its child components imports the styles */}
</Frame>
```
I have previously attempted making that a separate page and then have the content script do `<iframe src={chrome.runtime.getURL('page.html')}/>` but that runs into a lot of security issues. 

This is easy to do in a minimal repro if you want that.

**Reply by aklinker1** [reply]
Yeah, a small repro would be useful. So are the styles being generated from the react components imported?

**Reply by wallw-dev** [reply]
[Here is the reproduction](https://github.com/wallw-teal/vite-plugin-web-extension-iframe-css)

I think minimally just a hook like `(manifest: Manifest) => Promise<Manifest>` just after rendering the manifest and before validation would be fairly generic and allow some custom modifications. I'm happy to add that as a PR if you are interested.

**Reply by aklinker1** [reply]
That's what I was leaning toward as well. If you want to submit a PR, that'd be great. How does `transformManifest` sound?

**Reply by wallw-dev** [reply]
That works for me! PR is up at #123

**Reply by aklinker1** [reply]
Released in [v3.1.0](https://github.com/aklinker1/vite-plugin-web-extension/releases/tag/v3.1.0)
